### PR TITLE
C library: model pthread_key_create, pthread_{get,set}specific

### PR DESCRIPTION
**Pull request by tautschnig**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/cbmc/function-return-no-body1/test.desc [hunk 1]
```diff
@@ -5,7 +5,7 @@ activate-multi-line-match
 ^EXIT=10$
 ^SIGNAL=0$
 VERIFICATION FAILED
-<function_call hidden="false" step_nr="18" thread="0">\n\s*<function display_name="no_body" identifier="no_body">
-<function_return hidden="false" step_nr="19" thread="0">\n\s*<function display_name="no_body" identifier="no_body">
+<function_call hidden="false" step_nr="21" thread="0">\n\s*<function display_name="no_body" identifier="no_body">
+<function_return hidden="false" step_nr="22" thread="0">\n\s*<function display_name="no_body" identifier="no_body">
 --
```
Comment (kroening) [inline regression/cbmc/function-return-no-body1/test.desc]: Same here

Comment (tautschnig) [reply regression/cbmc/function-return-no-body1/test.desc]: This is gone now that #4270 is merged.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 729931299c1a6a670ac32ca34746afd6cd170697).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102155334
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: src/ansi-c/library/pthread_lib.c [hunk 2]
```diff
@@ -293,10 +293,23 @@ inline int pthread_mutex_destroy(pthread_mutex_t *mutex)
...
 
+extern __CPROVER_thread_local const void *__CPROVER_thread_keys[];
+extern __CPROVER_thread_local void (*__CPROVER_thread_key_dtors[])(void *);
+extern __CPROVER_thread_local unsigned long __CPROVER_next_thread_key;
+
 inline void pthread_exit(void *value_ptr)
 {
   __CPROVER_HIDE:;
   if(value_ptr!=0) (void)*(char*)value_ptr;
+#if 0
```
Comment (peterschrammel) [inline src/ansi-c/library/pthread_lib.c:304]: Suggest to add also a comment in the code why this code has been deactivated.

**Review by peterschrammel — APPROVED** [review]

File: src/ansi-c/library/pthread_lib.c (hunk 2 above)
Comment (tautschnig) [reply src/ansi-c/library/pthread_lib.c:304]: Done!

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: f9fee65f9315c7c3cd0e9701efa3c0f5a2b5a0f4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102970675
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

**Review by kroening — APPROVED** [review]

**Pull request merged**